### DISCUSSION TITLE
How does the RAG API's API key follow the user-provided in .env under custom endpoint conditions?

### DISCUSSION THREAD

**Discussion by sunsky89757** [danny-avila/LibreChat#2874] [Q&A]
### What is your question?

How does the RAG API's API key follow the user-provided in .env under custom endpoint conditions?

### More Details

I tried to use RAG_OPENAI_API_KEY=user_provided in .env, but it didn't work.
![image](https://github.com/danny-avila/LibreChat/assets/113254524/9c49ef08-cb76-4ec6-81c2-028534484d90)

Can this be designed to follow the RAG API with the user-provided API key?

### What is the main subject of your question?

Endpoints

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by sunsky89757** [comment]
STT_API_KEY and TTS_API_KEY are the same issue as RAG, I only take effect when I input a specific key like sk-xxx, and inputting user_provided is invalid

**Comment by danny-avila** [comment]
`user_provided` for the RAG_API is not supported yet

**Reply by dgrahn** [reply]
Does this continue to be the case?
